src/ucs/async/async.h

```diff
@@ -118,6 +119,24 @@ static inline int ucs_async_is_blocked(const ucs_async_context_t *async)
     return async->poll_block > 0;
 }
 
+/**
+ * Returns wether or not thread ownership has changed
+ * @param async Event context to check for ownership.
+ *
+ * @return 1 only if ownership is unchanged
+ */
+static inline int ucs_async_is_single_thread(ucs_async_context_t *async)
```

Comment:
maybe extend some existing unit test to verify the behavior?

Reply:
fixed